test/rewards/AutoCompoundingRewards.ts

```diff
@@ -686,6 +737,85 @@ describe('AutoCompoundingRewards', () => {
+                        { tokenSymbol: TokenSymbol.TKN, initialUserStake: toWei(30_000), totalRewards: toWei(13_000) },
+                        { tokenSymbol: TokenSymbol.TKN1, initialUserStake: toWei(40_000), totalRewards: toWei(14_000) },
+                        { tokenSymbol: TokenSymbol.TKN2, initialUserStake: toWei(50_000), totalRewards: toWei(15_000) },
+                        { tokenSymbol: TokenSymbol.TKN3, initialUserStake: toWei(60_000), totalRewards: toWei(16_000) },
+                        { tokenSymbol: TokenSymbol.TKN4, initialUserStake: toWei(70_000), totalRewards: toWei(17_000) },
+                        { tokenSymbol: TokenSymbol.TKN5, initialUserStake: toWei(80_000), totalRewards: toWei(18_000) }
+                    ];
+
+                    const tokens: TokenWithAddress[] = new Array<TokenWithAddress>(setups.length);
+                    const poolTokens: TokenWithAddress[] = new Array<TokenWithAddress>(setups.length);
+
+                    beforeEach(async () => {
+                        for (const [index, setup] of setups.entries()) {
+                            ({ token: tokens[index], poolToken: poolTokens[index] } = await prepareSimplePool(
+                                new TokenData(setup.tokenSymbol),
+                                setup.initialUserStake,
+                                setup.totalRewards
+                            ));
+
+                            await createProgram(
+                                distributionType,
+                                autoCompoundingRewards,
+                                tokens[index].address,
+                                setup.totalRewards,
+                                START_TIME
+                            );
+                        }
+
+                        await autoCompoundingRewards.setAutoProcessRewardsCount(AUTO_PROCESS_REWARDS_COUNT);
+                    });
+
+                    if (distributionType === RewardsDistributionType.Flat) {
+                        it('should distribute all tokens', async () => {
+                            await autoCompoundingRewards.setTime(programEndTime[distributionType]);
+                            for (let i = 0; i < Math.ceil(setups.length / AUTO_PROCESS_REWARDS_COUNT); i++) {
+                                const res = await autoCompoundingRewards.autoProcessRewards();
+                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                            }
+                            const res = await autoCompoundingRewards.autoProcessRewards();
+                            await expect(res).not.to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                        });
+
+                        it('should distribute some tokens', async () => {
+                            for (let i = 0; i < 5; i++) {
+                                await autoCompoundingRewards.setTime(
```

Comment:
I'm not sure if this achieves what we want if, but in general, it's unnecessary and too complicated (and also, you move the time to the end of all programs at the beginning, while we want to test previous distribution timestamps and also create "holes"). Basically, we need a simple test that:

1. Creates "discontinuous" triggering holes (X V V X X X V)
2. Tests that the hole is being properly skipped over. 

Can you simplify the test and make sure that it's properly tested? You don't need to go crazy and test all possible combinations, but instead, start with a single state (e.g., X V V X X X V), go over it in 1-2 loops (`autoProcessRewards()` cal,s) and make sure that the holes are being filled.

Reply:
Test updated.